api_public.go

```diff
@@ -27,57 +30,35 @@ type API struct {
 }
 
 // Handler returns the http.Handler instance for the API.
-func (api *API) Handler() http.Handler {
-	return api.router
+func (api API) Handler() http.Handler {
+	return api.router.Handler()
 }
 
-// Router can be used instead of Handler() to get the instance of julienschmidt httprouter.
-func (api *API) Router() *httprouter.Router {
+//Router returns the specified router on an api instance
+func (api API) Router() routing.Routeable {
 	return api.router
 }
 
-// AddResource registers a data source for the given resource
-// At least the CRUD interface must be implemented, all the other interfaces are optional.
-// `resource` should be either an empty struct instance such as `Post{}` or a pointer to
-// a struct such as `&Post{}`. The same type will be used for constructing new elements.
-func (api *API) AddResource(prototype jsonapi.MarshalIdentifier, source CRUD) {
```

Comment:
hm is there any reason you moved these 2 methods to the other file? They are public so why not let them be in `api_public.go`?